pkg/collector/systemperf_collector.go

```diff
@@ -22,23 +29,58 @@ func (collector *SystemPerfCollector) GetName() string {
 
 // Collect implements the interface method
 func (collector *SystemPerfCollector) Collect() error {
-	if err := utils.CreateKubeConfigFromServiceAccount(); err != nil {
-		return err
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Printf("Incluster config error: %v", err.Error())
 	}
 
-	output, err := utils.RunCommandOnContainer("kubectl", "top", "nodes")
+	metric, err := metrics.NewForConfig(config)
 	if err != nil {
+		log.Printf("Metrics for config error: %v", err)
+	}
+
+	// Node Metrics collector
+	nodeMetrics, err := metric.MetricsV1beta1().NodeMetricses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("Node metrics error:", err)
 		return err
 	}
 
-	collector.data["nodes"] = output
+	var nodeOutput strings.Builder
+	for _, nodeMetric := range nodeMetrics.Items {
+		cpuQuantity := nodeMetric.Usage.Cpu().MilliValue()
+		memQuantity, ok := nodeMetric.Usage.Memory().AsInt64()
+		if !ok {
+			return err
+		}
+		msg := fmt.Sprintf("Node Name: %s \n CPU usage: %d \n Memory usage: %d \n", nodeMetric.Name, cpuQuantity, memQuantity)
```

Comment:
Let's stick to objects as long as possible and marshall them to json (ex: https://github.com/Azure/aks-periscope/pull/90/files#diff-9d10783ad1196a63fa30faa58ec08cd3f2cfe2b8c6440f01a8962cf8359976cdR96) so that we have a structured output that can be easily parsed